src/plugins/intel_gpu/src/graph/program.cpp

```diff
@@ -1459,15 +1469,33 @@ std::pair<int64_t, int64_t> program::get_estimated_device_mem_usage() {
               [](program_node* const& lhs, program_node* const& rhs) {
                   return (lhs->get_output_layout().bytes_count() > rhs->get_output_layout().bytes_count());
               });
-
+    auto& engine = get_engine();
+    int64_t host_alloc = 0;
+    GPU_DEBUG_GET_INSTANCE(debug_config);
     // just to prevent the memories from being freed during allocation
     std::unordered_set<memory::ptr> allocated_mem_ptrs;
     for (const auto& node : nodes_to_allocate) {
         auto out_size = node->get_output_layout().bytes_count();
         if (out_size > max_alloc_size) {
             // to consider: if the base batch size is > 1, should we allow this single output allocation to host?
-            continue; // to be allocated to host
+            host_alloc += out_size;
+            continue;
         }
+        #ifdef __unix__
+        // Check whether the host mem allocation might exceed avialalbe system VRAM
+        int64_t total_host_alloc_size = out_size + host_alloc + engine.get_used_device_memory(allocation_type::usm_host);
+        if (engine.get_device_info().dev_type == cldnn::device_type::integrated_gpu)
+            total_host_alloc_size += engine.get_used_device_memory(allocation_type::usm_device);
+
+        if (cur_vmem != -1 && total_host_alloc_size > cur_vmem) {
```

Comment:
notice that LInux wouldn't allow to use the entire cur_vmem... maybe the conservative estimation would be a 0.5 of that, etc

Reply:
Actually at this step, it is prior to allocate current node. Also the "host_alloc" is not actually allocated too, and just being used for estimation. So I think it is enough to quit at this point.